crates/provider-http-server/tests/links.rs

```diff
@@ -0,0 +1,194 @@
+use std::sync::Arc;
+
+use anyhow::{Context as _, Result};
+use tokio::time::Duration;
+use tokio::try_join;
+
+use wasmcloud_test_util::control_interface::ClientBuilder;
+use wasmcloud_test_util::lattice::link::{assert_advertise_link, assert_remove_link};
+use wasmcloud_test_util::nats::wait_for_nats_connection;
+use wasmcloud_test_util::provider::StartProviderArgs;
+use wasmcloud_test_util::testcontainers::{AsyncRunner as _, ImageExt, NatsServer};
+use wasmcloud_test_util::{
+    assert_config_put, assert_scale_component, assert_start_provider, WasmCloudTestHost,
+};
+
+const PROVIDER_HTTP_SERVER_IMAGE_REF: &str = "ghcr.io/wasmcloud/http-server:0.23.2";
+const COMPONENT_HELLO_WORLD_RUST_IMAGE_REF: &str =
+    "ghcr.io/wasmcloud/components/http-hello-world-rust:0.1.0";
+const COMPONENT_HTTP_JSONIFY_RUST_IMAGE_REF: &str =
+    "ghcr.io/wasmcloud/components/http-jsonify-rust:0.1.1";
+
+/// Ensure duplicate links from the http provider to multiple components with
+/// the same details except link name act consistently
+///
+/// In this case, the flow of this regression test is roughly:
+///   - create provider & components
+///   - link the http server to *both* components (rust first)
+///   - Perform a HTTP request to the HTTP server
+///   - Delete the default link
+///   - The *second* established link (i.e. to the rust component) should still be present
+///   - The *third* established link (i.e. to the rust component) should still be present
+#[tokio::test]
+// TODO(vados-cosmonic): this test can be enabled once a new HTTP server is released
+#[ignore]
+async fn duplicate_link_shadowing() -> Result<()> {
+    // Build a wasmCloud host (assuming you have a local NATS server running)
+    let lattice = "default";
+    let component_hello_rust_id = "http-hello-rust";
+    let component_jsonify_rust_id = "http-jsonify-rust";
+    let provider_http_id = "http-server";
+    let provider_http_config_name = "http-server-config";
+    let provider_http_address = format!(
+        "127.0.0.1:{}",
+        wasmcloud_test_util::os::free_port_ipv4().await?
+    );
+    let endpoint_url = format!("http://{provider_http_address}");
+
+    // Start NATS
+    let nats_container = NatsServer::default()
+        .with_cmd(["--jetstream"])
```

Comment:
I wonder if we should just make this a default argument in the NatsServer?

JetStream seems like a fairly reasonable requirement every time, but I'm also totally fine keeping it just here if you think that'd be the best course of action for now, we can always revisit it if we see more examples where it would make sense.

Reply:
Ah this is a great idea, and I think you're right -- since this is the `testcontainers` feature provided by `wasmcloud-test-util`, it absolutely makes sense to add jetstream by default.

How do you feel about doing it as a follow up to this PR, would be short & sweet.

Reply:
Follow-up seems totally fine here 👍

Reply:
Filed an issue for this! https://github.com/wasmCloud/wasmCloud/issues/3490